src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -2154,14 +2150,12 @@ NOINLINE void ThawBase::recurse_thaw_interpreted_frame(const frame& hf, frame& c
 
   assert((stack_frame_bottom >= stack_frame_top + fsize) &&
          (stack_frame_bottom <= stack_frame_top + fsize + 1), ""); // internal alignment on aarch64
-
-  // on AArch64/PPC64 we add padding between the locals and the rest of the frame to keep the fp 16-byte-aligned
-  const int locals = hf.interpreter_frame_method()->max_locals();
   assert(hf.is_heap_frame(), "should be");
   assert(!f.is_heap_frame(), "should not be");
 
-  copy_from_chunk(heap_frame_bottom - locals, stack_frame_bottom - locals, locals); // copy locals
-  copy_from_chunk(heap_frame_top, stack_frame_top, fsize - locals);                 // copy rest
+  // Some architectures (like AArch64/PPC64/RISC-V) adds padding between the locals and the fixed_frame to keep the fp 16-byte-aligned.
+  // On those architectures we thaw the padding in order to keep the same localized pointer values.
+  copy_from_chunk(heap_frame_top, stack_frame_top, fsize);
 
   set_interpreter_frame_bottom(f, stack_frame_bottom); // the copy overwrites the metadata
```

Comment:
Since ThawBase::set_interpreter_frame_bottom has nothing to do after this change, I think it might be cleaner to remove this function at the same time?

Reply:
I see what you mean, but I chose to keep it because of the assert() in ThawBase::set_interpreter_frame_bottom.
After all, it was this assert that alerted me to the JDK-8305247 bug.

Reply:
Is it possible to get an equivalent but platform independent version of the assertion?
Something like `assert(f.interpreter_frame_local_at(0) == stack_frame_bottom - 1, "");` might work.
It could replace the call of `set_interpreter_frame_bottom()`.
After all with this pr no platform will ever have to actually set the interpreter frame bottom so it would be good to at least rename the method.

Reply:
Sounds like a plan. I'll look into it.

Reply:
Replaced the call to set_interpreter_frame_bottom in ThawBase::recurse_thaw_interpreted_frame with:
assert(f.interpreter_frame_local_at(0) == stack_frame_bottom - 1, "invalid frame bottom");
Then removed the set_interpreter_frame_bottom method from all platforms.